crates/diesel_models/src/query/invoice.rs

```diff
@@ -0,0 +1,59 @@
+use diesel::{associations::HasTable, BoolExpressionMethods, ExpressionMethods};
+use error_stack::report;
+
+use super::generics;
+use crate::{
+    errors,
+    invoice::{Invoice, InvoiceNew, InvoiceUpdate},
+    schema::invoice::dsl,
+    PgPooledConn, StorageResult,
+};
+
+impl InvoiceNew {
+    pub async fn insert(self, conn: &PgPooledConn) -> StorageResult<Invoice> {
+        generics::generic_insert(conn, self).await
+    }
+}
+
+impl Invoice {
+    pub async fn find_invoice_by_id(conn: &PgPooledConn, id: String) -> StorageResult<Self> {
+        generics::generic_find_one::<<Self as HasTable>::Table, _, _>(
+            conn,
+            dsl::id.eq(id.to_owned()),
+        )
+        .await
+    }
+
+    pub async fn find_invoice_by_merchant_id_invoice_id(
+        conn: &PgPooledConn,
+        merchant_id: &common_utils::id_type::MerchantId,
+        id: String,
+    ) -> StorageResult<Self> {
+        generics::generic_find_one::<<Self as HasTable>::Table, _, _>(
+            conn,
+            dsl::merchant_id
+                .eq(merchant_id.to_owned())
```

Comment:
we don't need merchant_id in query, since id is globally unique, if required the merchant_id can be a validation after it fetched and this need not to be in db